### PR TITLE
Use PV name as message key

### PR DESCRIPTION
**Pull request by matthew-d-jones**
### Issue

DM-1283

### Description of work

Uses the PV name as the message key for PV update messages published to Kafka. This will ensure the order of published PV messages is maintained.

Added a system test to cover this behaviour: `test_forwarding.py`, `test_updates_from_the_same_pv_reach_the_same_partition`

### Nominate for Group Code Review

- [ ] Nominate for code review

### CODE REVIEW

File: src/tests/tests.cpp [hunk 1]
```diff
@@ -1,6 +0,0 @@
-#include <gtest/gtest.h>
```
Comment (SkyToGround) [inline src/tests/tests.cpp]: I am assuming that test can be set to provide the `main()` function implementation? I would still like to have some comment on why this is (can be) removed.

Comment (matthew-d-jones) [reply src/tests/tests.cpp]: Yep, gtest provides its own `main()`. Only reason to define your own is if you want to have command line args or something like that.

Comment (SkyToGround) [reply src/tests/tests.cpp]: Does this mean that the standard gtest command line options (e.g. `--gtest_filter`) does not work anymore? If so, then I am not very happy with that change.

Comment (mattclarke) [reply src/tests/tests.cpp]: No they continue to work as before.

Comment (matthew-d-jones) [reply src/tests/tests.cpp]: @SkyToGround Here is the main it adds: https://github.com/google/googletest/blob/master/googletest/src/gtest_main.cc
It is identical to what we had, except an additional `printf`.

Comment (SkyToGround) [reply src/tests/tests.cpp]: Oh, ok then.

**Pull request merged**